### ISSUE TITLE
IMAPsync configfile configuration and generation

### ISSUE THREAD

**Issue by olaf7** [modoboa/modoboa#3245]
In any v2 version the IMAPmigration lacks the possibility to configure what needs to be migration resulting in the creation of a near .offlineimparc file which in turn errors the migration process.

I am going to add some screenshots v1 panel and v2 panel and comment below.

![image](https://github.com/modoboa/modoboa/assets/835414/f56317de-891b-4d1a-8b43-de14a6e57441)

![image](https://github.com/modoboa/modoboa/assets/835414/bd3d27ac-d69d-4b9d-8cc9-ae012ec999a6)

![image](https://github.com/modoboa/modoboa/assets/835414/3e08ffd2-5128-426c-9acc-f3ffad83e0fa)

![image](https://github.com/modoboa/modoboa/assets/835414/529e1d9a-bfa0-4689-9053-d28f112dbf94)

Problems:

-  no option to define host, port, domain, credentials (or other required or optional settings)
- mismatch between v1 and v2 panel settings (but this is the case in lots of settings in my instance). Here: enable imapsync, create folders
- when changing and saving  the setting to make imapsync active (enable) the change in v2 is not applied when checking the next time Iogin and check the setting

Resulting .offlineimaprc : 
`
[general]
accounts = 
`

**Comment by tonioo** [comment]
@olaf7 The synchronization issue is pretty strange and I don't reproduce it...
About offlineimap settings, once you have enabled the feature, you need to create at least one provider (gmail.com for example) where you'll be able to fill-in connection parameters and more.

**Comment by olaf7** [comment]
As in editing the almost empty config file provided by hand to suit my needs? Or will I be able to edit and or select anything from Modoboa/GUI once the config file is provided with a provider? A webform to supply details/parameters would be nice.

**Comment by tonioo** [comment]
Have you defined a provider and a migration yet?

**Comment by olaf7** [comment]
Please note: effectively this issue should be split into two issues imho. My bad. 1. mismatch settings v1/v2 and 2. IMAP migration
Also: can you comment on how I can help debug this as clearly something is wrong (as can be seen by screenshots), but you cannot reproduce.

To answer your question regarding the imap_sync issue: I created a .offlineimaprc based on the template, but it fails for a couple of reasons:

1.  IIRC no domain and accounts  should be created in advance. There is no existing domain, so no domain admin. Therefor I need to use Modoboa Administrator. This user has no postbox, so no Dovecot login, so offlineimap fails.
2. You do not know upfront what the remote IMAP server type is and what configuration is supported / what tweaking is required. Obviously I know what the server is in my case, but in general this is an issue. Settings can currently not be tweaked from within Modoboa.

After creation of the file nothing has changed to the GUI, so the screenshots provided here are still as it is. So editing is not possible. I assume this is by design?

Also note (more an OfflineIMAP issue, and something not easily to be avoided): passwords will be stored unencrypted in a file on disk.

**Comment by Spitfireap** [comment]
Please create another issue for the sync issue and keep this one for generation.

**Comment by xinomilo** [comment]
(sorry for the noize, but maybe the issue title and OP needs to change? this issue is about offlineimap, not imapsync., completely different things)

**Comment by tonioo** [comment]
@olaf7 Now that v1 is not there anymore, could you review the content of this issue and maybe create a new one?

**Comment by olaf7** [comment]
I can create new issues on request, but the following might suffice:

- Indeed there is no longer a difference between v1 and v2 panels as v1 is no longer in play
- When creating an IMAP migration there seem to be no longer a creation of an .offlineimaprc file as documented by: https://www.offlineimap.org/documentation.html so this no works differently or does it? IMAP migration does not work for me (correctly?) at the moment.
- The current IMAP provider definition allows for host, port, yes/no SSL, but offlineimap allows for other configurations as well. For instance does a server require a username in the format of 'user@example.net' or just plain 'user'.

**Comment by olaf7** [comment]
I agree this issue should be closed as it is no longer an issue as V1 is no more. However what remains is:
As long as a migration is defined, a new offlineimap configuration is created every time the cronjob is run (which is frequently).
It is unclear to me when (if?) at some point this file(structure) is deleted (from disk). Suggestion: please document this/the migration process.
Offlineimap allows for extensive configuration, however this cannot be tweaked from Modoboa.(as mentioned as third point in above comment and in the original subject) Changing things manually is useless as Modboa overwrites these changes.
As mentioned in another issue: passwords are in plaintext and keep reoccurring due to this process.
Probably these issues should be made into new issues so this one can be fully closed?
